apiserver/apiserver.go

```diff
@@ -846,7 +846,17 @@ func (srv *Server) endpoints() ([]apihttp.Endpoint, error) {
 			if err != nil {
 				return nil, nil, errors.Trace(err)
 			}
-			opener, err := resource.NewResourceOpener(st.State, store, srv.getResourceDownloadLimiter, tag.Id())
+			serviceFactory, err := httpCtxt.serviceFactoryForRequest(req)
+			if err != nil {
+				return nil, nil, errors.Trace(err)
```

Comment:
I would like to see this annotated. It is nice that we have stack traces but this is how we get 50 levels up in a call stack with just "not found" IMO.

Reply:
Something like "cannot get service factory for unit resource request"